src/Orchard/FileSystems/Media/FileSystemStorageProvider.cs

```diff
@@ -335,6 +352,17 @@ public class FileSystemStorageProvider : IStorageProvider {
             if (!Directory.Exists(dirName)) {
                 Directory.CreateDirectory(dirName);
             }
+            //Path.GetFileNameWithoutExtension(fileInfo.Name)
+            // If absolute path is longer than the maximum path length (260 characters), file cannot be saved.
+            if (fileInfo.FullName.Length > MaxPathLength) {
+                var fileName = Path.GetFileNameWithoutExtension(fileInfo.Name);
+                var extension = fileInfo.Extension;
+                // try to generate a shorter path for the file
+                var nameHash = fileName.GetHashCode().ToString("x");
```

Comment:
I think I already mentioned that but hashing is not stable. If the app restart the values might differ for the same input. There are stable hashes (SHA256, MD5) that can be used.